src/axolotl/integrations/nvfp4/kernels/bf16_fused_ce.py

```
@@ -0,0 +1,260 @@
+) -> bool:
+    """Patch ``model``'s ForCausalLM forward to use the chunked bf16 CE.
+
+    Returns True if a patch was installed (frozen bias-free nn.Linear lm_head).
+    Idempotent per ForCausalLM class; patching the base class covers PEFT too.
+    """
+    _set_vocab_block(vocab_block)
+    causal = model
+    if hasattr(model, "get_base_model"):
+        try:
+            causal = model.get_base_model()
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)